### PR TITLE
SPCTR-3245 : Encryption decryption logic added

### PR DESCRIPTION
**Pull request by rahulvarma722**
### Description

1. For Ai Api key validation added.
2. For Ai Api key Encryption and decryption logic added. 

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-admin-helper.php [hunk 1]
```diff
@@ -409,6 +409,57 @@ public static function get_global_content_width() {
...
+		 * Get open ai data.
+		 *
+		 * @since X.X.X
+		 * @return array
+		 */
+		public static function get_open_ai_data() {
+			$open_ai_options = self::get_admin_settings_option( 'uag_open_ai_options', array() );
+			
+			// Decrypt the key.
+			if ( is_array( $open_ai_options ) && ! empty( $open_ai_options['key'] ) ) {
```
Comment (bsf-zanev) [inline classes/class-uagb-admin-helper.php:423]: Should the ! empty check be first.

**Review by bsf-zanev — COMMENTED** [review]
A few comments.

File: classes/class-uagb-admin-helper.php [hunk 2]
```diff
@@ -409,6 +409,57 @@ public static function get_global_content_width() {
...
+
+		/**
+		 * Encrypt data using base64.
+		 *
+		 * @param string $input Input string which needs to be encrypted.
+		 * @since X.X.X
+		 * @return string
+		 */
+		public static function encrypt( $input ) {
+			if ( empty( $input ) || ! is_string( $input ) ) {
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply classes/class-uagb-admin-helper.php:438]: Just for security.

File: admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/utils.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    <svg className="animate-spin -mr-1 ml-3 h-5 w-5 text-white" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24">
+        <circle className="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" strokeWidth="4"></circle>
+        <path className="opacity-75" fill="currentColor" d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"></path>
+    </svg>
+);
+
+export async function authentiCateApiKey( openAiKey ) {
+	const messageArray = [{ role: 'user', content: 'Spectra Testing' }];
+
+	const response = await fetch( 'https://api.openai.com/v1/chat/completions', {
```
Comment (adi3890) [inline admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/utils.js:12]: can we try using /v1/usage api

File: admin-core/ajax/common-settings.php [hunk 4]
```diff
@@ -800,6 +800,13 @@ public function open_ai_options() {
 		$this->check_permission_nonce( 'uag_open_ai_options' );
 		$value = $this->check_post_value();
 		$value = json_decode( stripslashes( $value ), true );
+		$value = is_array( $value ) ? $value : array();
+
+		// Encrypt ai key.
+		if ( isset( $value['key'] ) && ! empty( $value['key'] ) ) {
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply admin-core/ajax/common-settings.php:806]: Done

File: admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/sections/OpenAIKey.js [hunk 5]
```diff
@@ -4,27 +4,18 @@ import ReactHtmlParser from 'react-html-parser';
 import { useDispatch } from 'react-redux';
 import { escapeHTML } from '@wordpress/escape-html';
 import getApiData from '@Controls/getApiData';
+import { svgSpinner, authentiCateApiKey } from '../utils';
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/sections/OpenAIKey.js:7]: Done

File: admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/sections/OpenAIKey.js [hunk 6]
```diff
@@ -4,27 +4,18 @@ import ReactHtmlParser from 'react-html-parser';
...
 
 const OpenAIKey = ( props ) => {
-	const {
-		openAIOptions,
-	} = props;
+	const { openAIOptions } = props;
 
-	// Decleration of all the states needed
+	// Deceleration of all the states needed
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/sections/OpenAIKey.js:12]: Done

File: classes/class-uagb-admin-helper.php (hunk 1 above)
Comment (rahulvarma722) [reply classes/class-uagb-admin-helper.php:423]: Done

File: admin-core/ajax/common-settings.php [hunk 7]
```diff
@@ -800,6 +800,12 @@ public function open_ai_options() {
 		$this->check_permission_nonce( 'uag_open_ai_options' );
 		$value = $this->check_post_value();
 		$value = json_decode( stripslashes( $value ), true );
+
+		// Encrypt ai key.
+		if( isset( $value[ 'key' ] ) && !empty( $value[ 'key' ] ) ){
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply admin-core/ajax/common-settings.php]: Removed

File: admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/sections/OpenAIKey.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/sections/OpenAIKey.js:12]: Done

File: admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/utils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/utils.js:12]: This function will only used in spectra ai that is why i have kept this in here.

Comment (rahulvarma722) [reply admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/utils.js:12]: Changed

File: admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/utils.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+// SVG For Right Hand Side Spinner.
+export const svgSpinner = (
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/utils.js:2]: @rahulvarma722 - Please add this SVG into `/scripts/pro-icons`

[the remark this answers is not in the document]
Comment (rahulvarma722) [reply admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/utils.js]: Added in SettingIcons.js file

File: admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/sections/OpenAIKey.js [hunk 9]
```diff
@@ -56,25 +50,24 @@ const OpenAIKey = ( props ) => {
...
 			security: uag_react.open_ai_options_nonce,
 			value: JSON.stringify( updatedOpenAIOptions ),
 		};
+
 		const getApiDataFetch = getApiData( {
 			url: uag_react.ajax_url,
 			action: 'uag_open_ai_options',
 			data : formData,	
 		} );
+
```
Comment (adi3890) [inline admin-core/assets/src/dashboard-app/pages/settings/spectra-ai-settings/sections/OpenAIKey.js:70]: can you please updae the text in line 75 as 
'Successfully saved!'

**Review by adi3890 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Pull request merged**